pkg/operator/customroute/custom_route_conditions.go

```diff
@@ -0,0 +1,197 @@
+package customroute
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"net/http"
+	"reflect"
+	"time"
+
+	configv1 "github.com/openshift/api/config/v1"
+	routev1 "github.com/openshift/api/route/v1"
+
+	"github.com/openshift/library-go/pkg/route/routeapihelpers"
+	"github.com/openshift/library-go/pkg/route/routecomponenthelpers"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+)
+
+func ensureDefaultConditions(conditions []metav1.Condition) []metav1.Condition {
+	for _, conditionType := range []string{"Progressing", "Degraded"} {
+		condition := findCondition(conditions, conditionType)
+		if condition == nil {
+			conditions = append(conditions, metav1.Condition{
+				Type:    conditionType,
+				Status:  metav1.ConditionFalse,
+				Reason:  "AsExpected",
+				Message: "All is well",
+			})
+		}
+	}
+	return conditions
+}
+
+func findCondition(conditions []metav1.Condition, conditionType string) *metav1.Condition {
+	for i := range conditions {
+		if conditions[i].Type == conditionType {
+			return &conditions[i]
+		}
+	}
+	return nil
+}
+
+func checkErrorsConfiguringCustomRoute(errors []error) []metav1.Condition {
+	if len(errors) != 0 {
+		return []metav1.Condition{
+			{
+				Type:    "Degraded",
+				Status:  metav1.ConditionTrue,
+				Reason:  "CustomRouteError",
+				Message: fmt.Sprintf("Error Configuring custom route: %v", errors),
+			},
+			{
+				Type:    "Progressing",
+				Status:  metav1.ConditionFalse,
+				Reason:  "CustomRouteError",
+				Message: fmt.Sprintf("Error Configuring custom route: %v", errors),
+			},
+		}
+	}
+	return nil
+}
+
+func checkIngressURI(ingressConfig *configv1.Ingress, route *routev1.Route) []metav1.Condition {
+	if _, _, err := routeapihelpers.IngressURI(route, route.Spec.Host); err != nil {
+		condition := &metav1.Condition{
+			Type:    "Progressing",
+			Status:  "True",
+			Reason:  "RouteNotAdmitted",
+			Message: fmt.Sprintf("Route not admitted: %v", err),
+		}
+		componentRoute := routecomponenthelpers.GetComponentRouteStatus(ingressConfig, "openshift-authentication", "oauth-openshift")
```

Comment:
(found while writing tests): this needs to point to the given component route namespace/name